Buildings/Fluid/Boilers/BoilerPolynomial.mo

```diff
@@ -1,262 +1,119 @@
+          Buildings.Utilities.Math.Functions.quadraticLinear(a=aQuaLin, x1=1, x2=T_nominal)
+        else
+          999);
+
+  parameter Buildings.Fluid.Types.EfficiencyCurves
+    effCur=Buildings.Fluid.Types.EfficiencyCurves.Constant
     "Curve used to compute the efficiency";
   parameter Real a[:] = {0.9} "Coefficients for efficiency curve";
-
-  parameter Buildings.Fluid.Data.Fuels.Generic fue "Fuel type"
-   annotation (choicesAllMatching = true);
-
-  parameter Modelica.SIunits.ThermalConductance UA=0.05*Q_flow_nominal/30
-    "Overall UA value";
-  parameter Modelica.SIunits.Volume VWat = 1.5E-6*Q_flow_nominal
-    "Water volume of boiler"
-    annotation(Dialog(tab = "Dynamics", enable = not (energyDynamics == Modelica.Fluid.Types.Dynamics.SteadyState)));
-  parameter Modelica.SIunits.Mass mDry =   1.5E-3*Q_flow_nominal
-    "Mass of boiler that will be lumped to water heat capacity"
-    annotation(Dialog(tab = "Dynamics", enable = not (energyDynamics == Modelica.Fluid.Types.Dynamics.SteadyState)));
-
-  Modelica.SIunits.Efficiency eta=
-    if effCur ==Buildings.Fluid.Types.EfficiencyCurves.Constant then
-      a[1]
-    elseif effCur ==Buildings.Fluid.Types.EfficiencyCurves.Polynomial then
-      Buildings.Utilities.Math.Functions.polynomial(a=a, x=y)
-   elseif effCur ==Buildings.Fluid.Types.EfficiencyCurves.QuadraticLinear then
-      Buildings.Utilities.Math.Functions.quadraticLinear(a=aQuaLin, x1=y, x2=T)
-   else
-      0
-  "Boiler efficiency";
-  Modelica.SIunits.Power QFue_flow = y * Q_flow_nominal/eta_nominal
-    "Heat released by fuel";
-  Modelica.SIunits.Power QWat_flow = eta * QFue_flow
-    "Heat transfer from gas into water";
-  Modelica.SIunits.MassFlowRate mFue_flow = QFue_flow/fue.h
-    "Fuel mass flow rate";
-  Modelica.SIunits.VolumeFlowRate VFue_flow = mFue_flow/fue.d
-    "Fuel volume flow rate";
-
-  Modelica.Blocks.Interfaces.RealInput y(min=0, max=1) "Part load ratio"
-    annotation (Placement(transformation(extent={{-140,60},{-100,100}})));
-
-  Modelica.Blocks.Interfaces.RealOutput T(final quantity="ThermodynamicTemperature",
-                                          final unit = "K", displayUnit = "degC", min=0)
-    annotation (Placement(transformation(extent={{100,70},{120,90}})));
-
-  Modelica.Thermal.HeatTransfer.Interfaces.HeatPort_a heatPort
-    "Heat port, can be used to connect to ambient"
-    annotation (Placement(transformation(extent={{-10,62}, {10,82}})));
-  Modelica.Thermal.HeatTransfer.Components.HeatCapacitor heaCapDry(
-    C=500*mDry,
-    T(start=T_start)) if not (energyDynamics == Modelica.Fluid.Types.Dynamics.SteadyState)
-    "heat capacity of boiler metal"
-    annotation (Placement(transformation(extent={{-80,12},{-60,32}})));
+  parameter Modelica.SIunits.Temperature T_nominal = 353.15
+    "Temperature used to compute nominal efficiency (only used if efficiency curve depends on temperature)";
```

Comment:
disable this parameter when the efficiency curve does not depend on temperature, so it will be
```
  parameter Modelica.SIunits.Temperature T_nominal = 353.15
    "Temperature used to compute nominal efficiency (only used if efficiency curve depends on temperature)"
    annotation (Dialog(enable= (effCur == Buildings.Fluid.Types.EfficiencyCurves.QuadraticLinear)));
```

Reply:
Change made in f506fe1.